app/models/service_template/copy.rb

```diff
@@ -0,0 +1,17 @@
+module ServiceTemplate::Copy
+  extend ActiveSupport::Concern
+
+  def copy(new_name = "Copy of " + name + Time.zone.now.to_s)
+    if template_valid? && type != 'ServiceTemplateAnsiblePlaybook'
+      ActiveRecord::Base.transaction do
+        dup.tap do |template|
+          template.update_attributes(:name => new_name, :guid => nil)
```

Comment:
Should `:guid` be a `SecureRandom.uuid`?

Reply:
This just resets it so that it will resave as a distinct SecureRandom.uui on https://github.com/ManageIQ/manageiq/blob/master/lib/uuid_mixin.rb#L10

Reply:
okay, but this still won't work for ```service template ansible tower``` because it depends on resources having ids and i don't, in the middle of the transaction. and tina's right, it seems pretty stupid to work around a single kind of template for that since that's the single case i run into in which that is going to be an issue.

Reply:
@gmcculloug @d-m-u I'm in favor of setting display to false.